airflow/models/dag.py

```diff
@@ -630,6 +630,7 @@ def get_next_data_interval(self, dag_model: "DagModel") -> Optional[DataInterval
         data_interval = dag_model.next_dagrun_data_interval
         if data_interval is not None:
             return data_interval
+        self.log.warning("%s needed to infer interval", self.dag_id)
```

Comment:
This was just my debugging/testing and we don't need this here I don't think.

Also it can stay if we think it's useful

Reply:
Happy to remove it then!